File: src/api/news-delete.ts
import type { NewsInformation, NewsItem, NewsSurvey, SessionHandle } from "~/models";
import { newsRemoteMutate } from "./private/news-remote-mutate";

/**
 * Will delete the item from the user's news feed.
 * @remark You can never get the item back after this.
 */
export const newsDelete = async (session: SessionHandle, item: NewsInformation | NewsSurvey): Promise<void> => {
  await newsRemoteMutate(session, item, {
    onlyMarkAsRead: false,
    markAsRead: false,
    delete: true
  });
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/api/news-delete.kt
package ink.literate.pawnote.api

import ink.literate.pawnote.api.private.NewsRemoteMutateOptions
import ink.literate.pawnote.api.private.newsRemoteMutate
import ink.literate.pawnote.models.NewsItem
import ink.literate.pawnote.models.SessionHandle

/** Will delete the item from the user's news feed. You can never get the item back after this. */
suspend fun newsDelete(session: SessionHandle, item: NewsItem<Any>) {
  newsRemoteMutate(
      session,
      item,
      NewsRemoteMutateOptions(onlyMarkAsRead = false, markAsRead = false, delete = true))
}
